### PR TITLE
Use SPARQL to speed up dbpedia data retrieval

### PR DESCRIPTION
**Pull request by davidbgk**
Refs etalab/udata#238

### CODE REVIEW

**Comment by MattiSG** [comment]
SPARQL and speedup in the same sentence?! :trollface:

File: dbpedia.py [hunk 1]
```diff
@@ -21,20 +38,36 @@ def __init__(self, resource):
...
-        return self.json
+        try:
+            return data['results']['bindings'][0][name]['value']
+        except IndexError:
+            pass
 
-    def __call__(self, *relations):
-        if not self.json and not self.fetch():
+        # Then fallback on the international one.
+        parameters['query'] = SPARQL_TEMPLATE.substitute(
```
Comment (noirbizarre) [inline dbpedia.py]: String template, yeah !

File: dbpedia.py [hunk 2]
```diff
@@ -21,20 +38,36 @@ def __init__(self, resource):
...
+            resource=resource,
+            ontology=DBPEDIA_ONTOLOGIES[name][1],
+            name=name)
+        response = requests.get(SPARQL_SERVER, params=parameters)
```
Comment (noirbizarre) [inline dbpedia.py]: Noob question: can't this two queries squashed into a single one ?

**Comment by noirbizarre** [comment]
r+

**Comment by davidbgk** [comment]
@MattiSG oh man, don't tempt me! ;p

File: dbpedia.py (hunk 2 above)
Comment (davidbgk) [inline dbpedia.py]: Probably, but I think that it's more future-proof to keep queries as simple and understandable as possible for people not familiar with semantic web concepts.

File: dbpedia.py [hunk 3]
```diff
@@ -1,7 +1,24 @@
...
+DBPEDIA_AREA = (
+    'http://fr.dbpedia.org/property/superficie',
+    'http://dbpedia.org/ontology/area'
+)
+DBPEDIA_ONTOLOGIES = {
+    'population': DBPEDIA_POPULATION,
+    'area': DBPEDIA_AREA,
+}
+SPARQL_SERVER = 'http://dbpedia.inria.fr/sparql'
+SPARQL_TEMPLATE = Template('select * where {<$resource> <$ontology> ?$name}')
```
Comment (MattiSG) [inline dbpedia.py]: Keywords in SPARQL are case-insensitive, but best practice is to uppercase them ([all examples from spec](http://www.w3.org/TR/rdf-sparql-query/#sparqlGrammar) are).

**Comment by davidbgk** [comment]
@noirbizarre @MattiSG new try with a unique SPARQL query, it looks to speedup even more the retrieval against INRIA's server… but increase complexity.

**Comment by noirbizarre** [comment]
r+²

**Pull request merged**